src/Factory/OrmFactory.php

```diff
@@ -26,6 +27,16 @@ public function __invoke(ContainerInterface $container)
             $container->get(CycleOrmHelper::class)->getCurrentSchemaArray(true, $this->params['generators'] ?? [])
         );
 
-        return (new ORM(new Factory($dbal)))->withSchema($schema);
+        $orm = (new ORM(new Factory($dbal)))->withSchema($schema);
+        // Promise factory
+        $promiseFactory = $this->params['promiseFactory'] ?? null;
+        if ($promiseFactory) {
+            if (!$promiseFactory instanceof PromiseFactoryInterface) {
+                $promiseFactory = $container->get($promiseFactory);
```

Comment:
```suggestion
                throw new \RuntimeException()
```

Reply:
Exception should be thrown automatically because `withPromiseFactory()` method requires `PromiseFactoryInterface` parameter